### PR TITLE
Fold global with non-integer scale factors

### PR DESCRIPTION
**Pull request by Qottmann**
Potentially easy fix for https://github.com/PennyLaneAI/pennylane/issues/3772
Had issues for local tests independent of this fix, trying here with CI

### CODE REVIEW

File: pennylane/transforms/mitigate.py [hunk 1]
```diff
@@ -33,7 +33,7 @@ def fold_global(circuit, scale_factor):
 
     .. math:: \text{fold_global}(U) = U (U^\dagger U)^n (L^\dagger_d L^\dagger_{d-1} .. L^\dagger_s) (L_s .. L_d)
 
-    where :math:`n = \lfloor (\lambda - 1)/2 \rfloor` and :math:`s = \lfloor \left((\lambda -1) \mod 2 \right) (d/2) \rfloor` are determined via the ``scale_factor`` :math:`=\lambda`.
+    where :math:`n = \lfloor (\lambda - 1)/2 \rfloor` and :math:`s = \lfloor \left((\lambda - 1) \right) (d/2) \rfloor` are determined via the ``scale_factor`` :math:`=\lambda`.
```
Comment (Qottmann) [inline pennylane/transforms/mitigate.py:36]:
```suggestion
    where :math:`n = \lfloor (\lambda - 1)/2 \rfloor` and :math:`s = \lfloor \left(\lambda - 1 \right) (d/2) \rfloor` are determined via the ``scale_factor`` :math:`=\lambda`.
```

**Comment by Qottmann** [comment]
Current failures are due to jax upgrade to 0.4.4 see https://github.com/PennyLaneAI/pennylane/pull/3784

File: tests/transforms/test_mitigate.py [hunk 2]
```diff
@@ -363,6 +363,24 @@ def qfunc(theta):
...
+        x = 0.5
+        circuit = qml.tape.QuantumScript([qml.RX(x, wires=0) for i in range(num_ops)])
+
+        d = len(circuit._ops)
+
+        folded, _ = qml.transforms.fold_global(circuit, lambda_)
+
+        n, s = divmod(lambda_ - 1, 2)
+
+        s = int(round(s * d / 2))
```
Comment (dwierichs) [inline tests/transforms/test_mitigate.py:379]:
```suggestion
        s = int(round(s * num_ops / 2))
```

File: tests/transforms/test_mitigate.py [hunk 3]
```diff
@@ -363,6 +363,24 @@ def qfunc(theta):
...
+
+        total_new = d * (2 * n + 1) + 2 * s
+        assert total_new == len(folded[0]._ops)
```
Comment (dwierichs) [inline tests/transforms/test_mitigate.py:382]:
```suggestion
        exp_total_new = d * (2 * n + 1) + 2 * s
        assert len(folded[0]._ops) == exp_total_new
```

File: tests/transforms/test_mitigate.py [hunk 4]
```diff
@@ -363,6 +363,24 @@ def qfunc(theta):
 class TestDifferentiableZNE:
     """Testing differentiable ZNE"""
 
+    @pytest.mark.parametrize("lambda_", [1.0, 1.5, 1.7, 2.0, 2.5])
+    @pytest.mark.parametrize("num_ops", [4, 8, 9, 12, 16])
+    def test_correct_number_of_operators(self, lambda_, num_ops):
+        """Test the output corresponds to the right number of operators according to https://arxiv.org/abs/2005.10921"""
```
Comment (dwierichs) [inline tests/transforms/test_mitigate.py:369]: Thanks for giving the reference, very useful :)
```suggestion
        """Test the output corresponds to the right number of operators according to Sec. II A 1) in  https://arxiv.org/abs/2005.10921"""
```

**Review by dwierichs — APPROVED** [review]
Thanks @Qottmann ! :100:  Just nit-picking a bit :)

File: tests/transforms/test_mitigate.py [hunk 5]
```diff
@@ -363,6 +363,24 @@ def qfunc(theta):
...
+        x = 0.5
+        circuit = qml.tape.QuantumScript([qml.RX(x, wires=0) for i in range(num_ops)])
+
+        d = len(circuit._ops)
```
[the remark this answers is not in the document]
Comment (Qottmann) [reply tests/transforms/test_mitigate.py:373]: Yes, but I got a bit paranoid playing around with QuantumScript

[the remark this answers is not in the document]
Comment (Qottmann) [reply tests/transforms/test_mitigate.py]: removed it 👍

[the remark this answers is not in the document]
Comment (dwierichs) [reply tests/transforms/test_mitigate.py]: Totally relatable :D Then we should do `assert len(circuit._ops)==num_ops` maybe?

**Review by AlbertMitjans — APPROVED** [review]
Thanks!!

**Comment by timmysilv** [comment]
thank you to the reviewers for beating me to it!

**Pull request merged**